packages/astro/src/core/preview/index.ts

```diff
@@ -66,16 +66,19 @@ export default async function preview(inlineConfig: AstroInlineConfig): Promise<
 	if (typeof previewModule.default !== 'function') {
 		throw new Error(`[preview] ${settings.adapter.name} cannot preview your app.`);
 	}
-
+	const previewConfig = Object.assign({},
+		settings.config.vite.server,
+		settings.config.vite.preview,
+		settings.config.server,
+	)
```

Comment:
According to the [official Vite documentation](https://vitejs.dev/config/preview-options.html),  `vite preview` uses the Vite Server configuration by default. Astro is built on Vite, but doesn't use Vite's Server configuration, which is very strange. (The proxy configuration in Dev and Preview is very useful for this purpose.

Anyway, the above code just keeps Astro Preview behavior consistent with Vite Preview, and is technically a bug fix.

Reply:
Astro preview does use the Vite preview server internally, but we don't document that the Vite server/preview options will strictly work when configured. Editing this part of the code is also sensitive because you're passing all the Vite-specific server/preview options to `previewModule.default` directly.

We're currently intentionally limiting the options here because it would be expected for all adapters to support the same set of features, but not all adapters use the Vite preview server. That way, the adapters need to only concern of supporting features from `config.server`.

---

If this is about the `proxy` config, it leans towards a feature request, not a bug fix. I could see perhaps passing that only to `previewModule.default` and update the types to be optional, and adapters could decide to support it. But I'd like to hear what others think of this.

Reply:
Got it. What I need is `proxy`, it can be approached by configuring vite.server on astro dev, but doesn't work on astro preview.